### PR TITLE
Cleanup & Fixes: derives, output formatting, Dynamic section parsing fixed

### PR DESCRIPTION
**Pull request by globin**
Please check each commit for the different changes, if there are any controversial changes I'll gladly open PRs for the uncontroversial ones to get them merged more easily.

### CODE REVIEW

File: src/hash.rs [hunk 1]
```diff
@@ -33,9 +33,9 @@ impl HashTable {
         }
     }
 
-    pub fn get_chain(&self, index: u32) -> u32 {
-        assert!(index < self.chain_count);
-        let index = self.bucket_count + index;
+    pub fn get_chain(&self, chain_index: u32) -> u32 {
+        assert!(chain_index < self.chain_count);
+        let index = self.bucket_count + chain_index;
```
Comment (nrc) [inline src/hash.rs]: This doesn't seem like a good chain - it just adds duplication to the name and doesn't make anything clearer

**Review by nrc — APPROVED** [review]
Thank you! Mostly looks great, just a rename I don't like.

**Comment by globin** [comment]
Fixed

**Comment by nrc** [comment]
Thanks!

**Pull request merged**